digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/Digiroad2Api.scala

```diff
@@ -259,31 +259,32 @@ class Digiroad2Api(val roadLinkService: RoadLinkService,
           Map("success" -> false)
       }
     }
-  /**
-  Returns empty result as Json message, not as page not found
-  */
-  get("/massTransitStopsSafe/:nationalId") {
-    def validateMunicipalityAuthorization(nationalId: Long)(municipalityCode: Int): Unit = {
+
+  get("/massTransitStops/passenger/:passengerId") {
+    def validateMunicipalityAuthorization(passengerId: String)(municipalityCode: Int): Unit = {
       if (!userProvider.getCurrentUser().isAuthorizedToRead(municipalityCode))
-        halt(Unauthorized("User not authorized for mass transit stop " + nationalId))
+        halt(Unauthorized("User not authorized for mass transit stop " + passengerId))
     }
-    val nationalId = params("nationalId").toLong
-    val massTransitStopReturned = massTransitStopService.getMassTransitStopByNationalIdWithTRWarnings(nationalId, validateMunicipalityAuthorization(nationalId))
-    massTransitStopReturned._1 match {
-      case Some(stop) =>
-        Map("id" -> stop.id,
-          "nationalId" -> stop.nationalId,
-          "stopTypes" -> stop.stopTypes,
-          "lat" -> stop.lat,
-          "lon" -> stop.lon,
-          "validityDirection" -> stop.validityDirection,
-          "bearing" -> stop.bearing,
-          "validityPeriod" -> stop.validityPeriod,
-          "floating" -> stop.floating,
-          "propertyData" -> stop.propertyData,
-          "success" -> true)
-      case None =>
-        Map("success" -> false)
+    val passengerId = params("passengerId")
+    val massTransitStopsReturned = massTransitStopService.getMassTransitStopByPassengerId(passengerId, validateMunicipalityAuthorization(passengerId))
+    massTransitStopsReturned.map { massTransitStopReturned =>
+      massTransitStopReturned match {
+        case Some(stop) =>
+          Map("id" -> stop.id,
+            "nationalId" -> stop.nationalId,
+            "stopTypes" -> stop.stopTypes,
+            "lat" -> stop.lat,
+            "lon" -> stop.lon,
+            "validityDirection" -> stop.validityDirection,
+            "bearing" -> stop.bearing,
+            "validityPeriod" -> stop.validityPeriod,
+            "floating" -> stop.floating,
+            "propertyData" -> stop.propertyData,
+            "municipalityName" -> stop.municipalityName.getOrElse(""),

```

Comment:
We could send only the national Id, lon, lat, or whatever is needed for the search and add the others later if needed.